### PR TITLE
Token Drain Status Polling Endpoint

### PR DESCRIPTION
**Pull request by husobee**
### Summary

This PR enables a caller (client) to receive a drain id from submission of tokens, and provides the ability to query a new get drain poll endpoint to look up the status of the draining request.

### Type of change ( select one )

- [ ] Product feature
- [ ] Bug fix
- [x] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [x] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [x] Have you squashed all intermediate commits?
- [x] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: migrations/0024_drain_poll.up.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+--- completed indicates this claim_drain job is drained and complete
+alter table claim_drain add column completed boolean not null default false;
+--- completed_at indicates the time at which this claim_drain job was completed
+alter table claim_drain add column completed_at timestamp;
+--- batch_id is the draining batch that this claim drain job belongs to
+alter table claim_drain add column batch_id uuid default null;
+--- create an index on the batch_id for easy lookup
+create index batch_id_idx on claim_drain(batch_id);
```
Comment (maikelmclauflin) [inline migrations/0024_drain_poll.up.sql:8]: i definitely like batches better

Comment (husobee) [reply migrations/0024_drain_poll.up.sql:8]: :+1:

File: promotion/datastore.go [hunk 2]
```diff
@@ -628,6 +633,73 @@ func (pg *Postgres) SaveClaimCreds(creds *ClaimCreds) error {
...
+		err       error
+	)
+
+	statement := `
+select
+	batch_id,
+	bool_and(completed) as completed,
+	bool_or(erred) as delayed,
+	(not bool_and(completed) and not bool_or(erred)) as inprogress,
+	(not bool_or(completed)) as pending
```
Comment (maikelmclauflin) [inline promotion/datastore.go:656]: would this line cause the initial state to be `unknown`?

Comment (husobee) [reply promotion/datastore.go:656]: I think unknown state is impossible to achieve, as if there is a record, either completed/delayed/inprogress/pending would return a true value.  The unknown fail-safe should not be reached imo.

Comment (husobee) [reply promotion/datastore.go:656]: tests showing how this unknown state are achieved have been added.  https://github.com/brave-intl/bat-go/pull/654/commits/d1d99642493fd663e8d5971aea7527d6171befb0

**Review by maikelmclauflin — DISMISSED** [review]

**Review by maikelmclauflin — APPROVED** [review]

**Pull request merged**